### ISSUE TITLE
"App needs to be updated" on IOS 15.6

### ISSUE THREAD

**Issue by m50434** [nspire-emus/firebird#266]
Hi there,
im using an ipad (8. gen.) with ios 15.6.

If I try to open the emulator app, it says "App needs to be updated".

What can I do?

Thank you very much.

**Comment by adriweb** [comment]
> What can I do?

Well, _you_ can rebuild the app locally... if you can't wait, I suppose, but otherwise you'll need to wait for a new release I guess :P (or at least a pre-release with the iOS binary)
That said, I'm not even sure why this message pops up, it's possible the new OS wants apps to be built with some recent enough toolchain for various reasons now...

**Comment by m50434** [comment]
Hey,
thanks for your fast reply.

I think I cant rebuild it by myself because I have now idea how to :-( 

> otherwise you'll need to wait for a new release I guess

The last release is from march 2020. So I guess I have to wait a long time?

**Comment by adriweb** [comment]
I'm not sure, I may have some time to take care of some things this week, in fact...

**Comment by m50434** [comment]
Me again:
I got another Ipad ;-) with iOS 14.3

Now, I got another error: "The provisioning profile" is expired. Profile expiration "2021-02-18 21:08:28".

So there seems to be another problem.

**Comment by adriweb** [comment]
Yes that would make sense, whichever binary I will build and upload here will use the latest unexpired stuff.

**Comment by m50434** [comment]
So, this error could be resolve as well with a new release?

**Comment by adriweb** [comment]
Yes.

Basically Apple doesn't really want people to release things out of the app store, so it comes with a whole set of limitations (I'm not even sure how this currently "works", even), so normally it would be up to users to "just" build their own version (that's free, no dev subscription needed) and there would be no issue, but of course I understand it's annoying to set up.

**Comment by adriweb** [comment]
Ok, so, I have NOT been able to test this, but this is a release build from the latest master 🤷‍♂
[Firebird-Emu_iOS_Release_master_20220804.ipa.zip](https://github.com/nspire-emus/firebird/files/9258709/Firebird-Emu_iOS_Release_master_20220804.ipa.zip)
️

**Comment by m50434** [comment]
Thank you very much.
Unfortunately it shows the same errors on both devices :-(

**Comment by adriweb** [comment]
Hmm ok then the problem is something else :(
I'll have to dig a bit deeper

**Comment by m50434** [comment]
Me again.
I tried it again, with both of my test-devices (because last time I was a bit in a hurry).
The error "App needs to be updated" on iOS 15.6 is gone.
But on both devises there is now the error "the app cannot be installed because its integrity could not be verified"

![grafik](https://user-images.githubusercontent.com/24473301/183242415-c4cfef69-7ebc-417b-b2ea-888dcddfcbfc.png)

**Comment by adriweb** [comment]
Ok I suppose that does make sense now considering the adhoc signing isn't supposed to work that way (but its what I've always done so I assume users just had jailbroken devices)

If you can download Xcode and Qt (5.15) I can let you know how you can build it yourself and it'll run fine on your devices

**Comment by m50434** [comment]
Unfortunately I have no MAC and therefore no possibility to install XCode.

But thank you very much!

**Comment by adriweb** [comment]
Maybe you could use the web version of Firebird then 😅

**Comment by m50434** [comment]
oh, where can I find this?

**Comment by m50434** [comment]
Hey,
finally I got it working (with the help of "Sideloadly").
Now I have the next problem: Im unable to chose a "store-point" because there is only a dialog, which shows the intern app files.
In Itunes I didnt see any "file sharing" for the app.

Any ideas?

**Comment by adriweb** [comment]
Oh glad that this is working, should have thought of that.

Regarding the file sharing, I may have forgotten to enable that after cleaning up some metadata oops. 

In the meantime you might be able to choose files downloaded by safari?

**Comment by m50434** [comment]
no there is no Safari option. I can only See the App file system
![Uploading 7FCCB289-79B8-4B3E-8DB7-9C19E51AAD1C.png…]()

**Comment by adriweb** [comment]
Ah okay I guess I'll have to fix that for real then. 
if only Qt used the native file picker... 👀

**Comment by m50434** [comment]
Me again and finally:
Its working now :-) :-)
In sideloadly there is a option for "UIFileSharing". So I could add the Flahs-Boot-Store-Files via Itunes to the app. 

By the way: It also works with your last release from March 2020. So the errors I describes a few days ago aren't there anymore (sideloady did the trick).

Anyway: Thank you very much for your (fast) support!

**Comment by Aaron668-5921** [comment]
Just wanted to leave a note that I am unfortunately limited to web based ipa installing and as experienced m above, same errors.  My need is not urgent so no rush, but as you’re able the updates you were working on would still be greatly appreciated.

Thank you for the work you’ve done already. ☺️🙏

**Comment by adriweb** [comment]
Well I'm not able to provide more than what I've tried so far, actually :(
So I suppose either a jailbroken device or he local ipa installing tricks are required...

**Comment by Aaron668-5921** [comment]
Ahh, understand.  No worries.  Thank you again for what you have already done.

**Comment by adriweb** [comment]
That's great !
Thanks for the trick, I guess we'll mention it somewhere :)
